src/toniarts/openkeeper/world/creature/steering/AbstractCreatureSteeringControl.java

```diff
@@ -105,19 +102,27 @@ protected void applySteering(SteeringAcceleration<Vector2> steering, float tpf)
         // TODO: Call function?
         if (steering.isZero()) {
             steeringBehavior = null;
+            stopMoving = false;
+            return;
         }
 
         // Update orientation and angular velocity
-        if (independentFacing) {
-            setOrientation(getOrientation() + (angularVelocity * tpf));
-            angularVelocity += steering.angular * tpf;
+        if (!stopMoving) {
+            position.add(linearVelocity.x * tpf, linearVelocity.y * tpf);
+            linearVelocity.mulAdd(steering.linear, tpf).limit(maxLinearSpeed);

```

Comment:
Are these 2 lines are really necessary? They are getting called some lines above, so if it runs into this block it is called twice.

At least my tests made them move like before (without the weird spinning)